### DISCUSSION TITLE
Block Adsense Cookies

### DISCUSSION THREAD

**Discussion by Oleyoytutambien** [orestbida/cookieconsent#618] [General]
How do i block Adsense cookies?. I 've tryed the same code than Analytics, but isn't working for Adsense.
I'm trying this:
```
<script async src="https://www.googletagmanager.com/gtag/js?id=G-XXXXXXXX"></script>
<ins class="adsbygoogle"
     style="display:inline-block;width:XXXpx;height:XXXpx"
     data-ad-client="ca-pub-xxxxxxxxxxxxxxx"
     data-ad-slot="xxxxxxxxxx">
</ins>
<script type="text/plain" data-category="marketing" data-service="Google Adsense">
	(adsbygoogle = window.adsbygoogle || []).push({});
</script>
```
But when i do this:
1.- Adsense ads are not showing, even if you accept everything, clean all cookies, reload several times...
2.- Even denied all the cookies, __gads and __gpi cookies are saved.

Please, any idea about how to fix it?.
Thank you in advance!

**Comment by orestbida** [comment]
@Oleyoytutambien 

> Beginning January 16, 2024, publishers and developers using Google AdSense, Ad Manager, or AdMob will be required to use a Consent Management Platform (CMP) that has been certified by Google and has integrated with the [IAB's Transparency and Consent Framework](https://iabeurope.eu/transparency-consent-framework/) (TCF) when serving ads to users in the European Economic Area or the UK. 

[(source)](https://support.google.com/adsense/answer/13554020?hl=en)

If you still want to try implementing it, this is how I would do it:

```html
<html>
    <head>
        <!-- head content-->
        <!-- plugin css -->
    </head>
    <body>
        <!-- body content -->
        <!-- plugin script -->

        <ins class="adsbygoogle"
            style="display:inline-block;width:728px;height:90px"
            data-ad-client="ca-pub-1234567890123456"
            data-ad-slot="1234567890"
        ></ins>

        <script 
            type="text/plain" 
            data-category="marketing" 
            data-service="Google Adsense"
        >
            (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
        <script 
            type="text/plain"
            data-category="marketing"
            data-service="Google Adsense"
            async
            src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js?client=ca-pub-1234567890123456" 
            crossorigin="anonymous"
        ></script>
    </body>
</html>
```

The above code was adapted from the [official snippet](https://support.google.com/adsense/answer/9190028?hl=en).

**Reply by Oleyoytutambien** [reply]
Thank you for your help. I did just as you said, but it's not working for me. It works perfect with Analytics, but not with Adsense. Maybe i have to find another way. If i don't find a solutiong, I will try to block the whole page if they don't accept the cookies. This way, i don't need to change anything with Adsense.
